### PR TITLE
Edited how to run dataset based tests notebook

### PR DESCRIPTION
**Pull request by validbeck**
## Internal Notes for Reviewers

Edits made on this notebook:

- CTA for the title
- Revised introduction with conceptual breakdown
- A new table of contents
- Revised "About ValidMind" section
- A new "Key concepts" section
- Revised "Next steps" footer

I also made some slight edits to the instructions for clarity and adherence to our brand style guide, including some definitions for terms that pop up along the way and more granular section breakdowns.

### CODE REVIEW

File: notebooks/archive/configure_test_parameters.ipynb
Comment (cachafla) [inline notebooks/archive/configure_test_parameters.ipynb:1]: What happened to this notebook?

Comment (validbeck) [reply notebooks/archive/configure_test_parameters.ipynb:1]: Screenshot 2024-04-30 at 2 59 58 PM

I moved it into an [`archive` folder](https://github.com/validmind/developer-framework/tree/beck/sc-4154/edit-how-to-run-dataset-based-tests-notebook/notebooks/archive) as Nik said this new notebook replaces the following: 

Screenshot 2024-04-30 at 3 00 29 PM

If this is incorrect I can make the adjustments accordingly!

Comment (cachafla) [reply notebooks/archive/configure_test_parameters.ipynb:1]: Ah yeah, I remember now 😬

**Review by cachafla — APPROVED** [review]

**Review by nrichers — APPROVED** [review]
This looks great, Beck! Very nice work on this notebook edit. 🚀 

One thing to be aware of: notebook renames can break our e2e test notebook, but this [does not appear to be one of the notebooks](https://github.com/validmind/developer-framework/blob/beck/sc-4154/edit-how-to-run-dataset-based-tests-notebook/scripts/run_e2e_notebooks.py#L40-L51) that gets tested. (If it were, you would need to update the e2e notebook with the new name.)

**Pull request merged**